File: AscendDev.Core/Models/Social/Discussion.cs
using AscendDev.Core.Models.Auth;
using AscendDev.Core.Models.Courses;

namespace AscendDev.Core.Models.Social;

public class Discussion
{
    public Guid Id { get; set; }
    public string? LessonId { get; set; }
    public string? CourseId { get; set; }
    public Guid UserId { get; set; }
    public string Title { get; set; } = null!;
    public string Content { get; set; } = null!;
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public bool IsPinned { get; set; } = false;
    public bool IsLocked { get; set; } = false;
    public int ViewCount { get; set; } = 0;

    // Navigation Properties
    public User User { get; set; } = null!;
    public Lesson Lesson { get; set; } = null!;
    public List<DiscussionReply> Replies { get; set; } = new();
    public List<DiscussionLike> Likes { get; set; } = new();

    // Computed Properties
    public int ReplyCount => Replies.Count;
    public int LikeCount => Likes.Count;
    public DateTime LastActivity => Replies.Any() ?
        Replies.Max(r => r.UpdatedAt ?? r.CreatedAt) :
        UpdatedAt ?? CreatedAt;
}

File: frontend/src/types/discussion.ts
export interface User {
    id: string;
    username: string;
    email: string;
    profilePictureUrl?: string;
}

export interface Discussion {
    id: string;
    lessonId?: string;
    courseId?: string;
    userId: string;
    title: string;
    content: string;
    createdAt: string;
    updatedAt?: string;
    isPinned: boolean;
    isLocked: boolean;
    viewCount: number;
    replyCount: number;
    likeCount: number;
    isLikedByCurrentUser: boolean;
    lastActivity: string;
    user: User;
    replies?: DiscussionReply[];
}

export interface DiscussionReply {
    id: string;
    discussionId: string;
    userId: string;
    content: string;
    createdAt: string;
    updatedAt?: string;
    parentReplyId?: string;
    isSolution: boolean;
    isEdited: boolean;
    depth: number;
    user: User;
    childReplies?: DiscussionReply[];
}

export interface CreateDiscussionRequest {
    title: string;
    content: string;
    lessonId?: string;
    courseId?: string;
}

export interface UpdateDiscussionRequest {
    title?: string;
    content?: string;
    isPinned?: boolean;
    isLocked?: boolean;
}

export interface CreateDiscussionReplyRequest {
    content: string;
    parentReplyId?: string;
}

export interface UpdateDiscussionReplyRequest {
    content: string;
    isSolution?: boolean;
}